base/loading.jl

```diff
@@ -3308,6 +3316,65 @@ function resolve_depot(inc::AbstractString)
     return :no_depot_found
 end
 
+"""
+    RelocPath(path::AbstractString)
+
+A type to represent a relocatable path.
+
+Requires `path` to be located within one of `DEPOT_PATH` upon construction.
+
+An error is thrown if relocation fails.
+
+# Example
+```jldoctest
+julia> path1 = joinpath(mktempdir(), "foo"); touch(path1); # set up a file called foo
+
+julia> pushfirst!(DEPOT_PATH, dirname(path1));
+
+julia> relocpath = RelocPath(path1);
```

Comment:
Doctests are failing because `RelocPath` is not exported from Base. We either need to export it, or fix the doctest.

```suggestion
julia> relocpath = Base.RelocPath(path1);
```